src/gax/src/retry_policy.rs

```diff
@@ -395,9 +421,8 @@ where
         if now < deadline {
             None
         } else {
-            Some(Error::other(format!(
-                "limited time retry policy exhausted {:?} ago",
-                now.saturating_duration_since(deadline)
+            Some(Error::exhausted(LimitedElapsedTimeError::new(
+                self.maximum_duration,
             )))
```

Comment:
Now that I think about it, doesn't this drop any service errors from the retry loop? Can we surface the last failure?